### ISSUE TITLE
IB problems in generation and loss computation

### ISSUE THREAD

**Issue by luluisthebest** [Xiaoqiang-Yan/DCIB#2]
您好，非常感谢您的工作，对我的启发很大，但是我有几点疑问：
1. 每个瓶颈层最后都有ReLU()，导致mu, sigma = params[:, :self.feature_dim], params[:, self.feature_dim:]两个参数都为非负，但是均值应该是任意实数，这边没太理解您的设计？
2. 计算互信息时，kl_div函数的用法好像不正确，它应该是计算两个分布的，不是计算两个采样样本的，这边也没太理解您的用意？

> def getMILoss(P_F, P, mi_estimator):
    x_P_F = P_F.rsample()
    prior_sample = prior.sample().to(device)
    loss1_1 = 0
    loss2 = 0
    loss3 = 0
    for p in P:
        x_p = p.rsample()
        miG, _ = mi_estimator(x_P_F, x_p)
        loss2 += mutual_information(x_p, x_P_F)
        loss1_1 += -miG
        loss3 += torch.nn.functional.kl_div(x_p, prior_sample).to(device)
    return loss1_1, loss2, loss3

Thank you very much for your work. It has been very inspiring to me. However, I have a few questions:
1. At the end of each bottleneck layer, there is a ReLU(), which makes both mu and sigma = params[:, :self.feature_dim], params[:, self.feature_dim:] non-negative. But the mean should be any real number. I'm not quite clear about your design here?
2. When calculating mutual information, the usage of the kl_div function seems incorrect. It should be used to calculate two distributions, not two sampled samples. I'm also not quite clear about your intention here?
